common/src/chain/transaction/output/output_value.rs

```diff
@@ -34,6 +45,13 @@ impl OutputValue {
             OutputValue::TokenV0(_) | OutputValue::TokenV1(_, _) => None,
         }
     }
+
+    pub fn amount(&self) -> Amount {
+        match self {
+            OutputValue::Coin(v) | OutputValue::TokenV1(_, v) => *v,
+            OutputValue::TokenV0(_) => panic!("deprecated token version"),
+        }
+    }
```

Comment:
This function feels like playing with fire to me. I think it should go. 

If you need something like this, make it local to where you're using it.